laboratory-servlet-api-integration/src/main/java/com/wixpress/petri/laboratory/http/LaboratoryFilter.java

```diff
@@ -121,6 +134,26 @@ public void init(FilterConfig filterConfig) throws ServletException {
         startMetricsReporterScheduler(laboratoryTopology.getReportsScheduleTimeInMillis());
 
         FilterTypeIdResolver.useDynamicFilterClassLoading();
+        cache = setupKoboshiCache(context);
+    }
+
+    private ReadOnlyTimestampedLocalCache<ConductibleExperiments> setupKoboshiCache(final ServletContext context) {
```

Comment:
yeah, this Filter is indeed a mess.
but this specific area was already extracted into the (now deleted) PetriClientExperimentSource.
so I feel boy scout rule should apply here..

also, I have a feeling the whole setup part might benefit from being further encapsulated by koboshi, and having this in a separate class might help show that / test that.